scripts/build/gen_handles.py

```diff
@@ -372,39 +372,52 @@ def main():
             inj_paths = []
             sup_paths = []
             hdls = []
+            hdls_temp = []
 
             sn = hs.node
+            hdls_temp.clear()
             if sn:
-                hdls.extend(dn.__device.dev_handle for dn in sn.__depends)
+                hdls_temp.extend(dn.__device.dev_handle for dn in sn.__depends)
+                hdls_temp.sort()
+                hdls.extend(hdls_temp)
                 for dn in sn.depends_on:
                     if dn in sn.__depends:
                         dep_paths.append(dn.path)
                     else:
                         dep_paths.append('(%s)' % dn.path)
             # Force separator to signal start of injected dependencies
             hdls.append(DEVICE_HANDLE_SEP)
+            hdls_temp.clear()
             for inj in hs.dev_injected:
                 if inj not in edt.dep_ord2node:
                     continue
                 expected = edt.dep_ord2node[inj]
                 if expected in used_nodes:
                     inj_paths.append(expected.path)
-                    hdls.append(expected.__device.dev_handle)
+                    hdls_temp.append(expected.__device.dev_handle)
+                hdls_temp.sort()
+                hdls.extend(hdls_temp)
```

Comment:
I think this will break, the call to `clear` is above the for loop and you call `sort` and `extend` inside. If the loop runs multiple times you'll get duplicates

Reply:
This comment no longer applies to the latest version of the PR.